src/protagonist/API.Tests/Integration/CustomHeaderTests.cs

```diff
@@ -0,0 +1,200 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using API.Client;
+using API.Tests.Integration.Infrastructure;
+using DLCS.Model.Assets.CustomHeaders;
+using DLCS.Repository;
+using Hydra.Collections;
+using Microsoft.EntityFrameworkCore;
+using Test.Helpers.Integration;
+using Test.Helpers.Integration.Infrastructure;
+
+namespace API.Tests.Integration;
+
+[Trait("Category", "Integration")]
+[Collection(CollectionDefinitions.DatabaseCollection.CollectionName)]
+public class CustomHeaderTests : IClassFixture<ProtagonistAppFactory<Startup>>
+{
+    private readonly HttpClient httpClient;
+    private readonly DlcsContext dlcsContext;
+
+    public CustomHeaderTests(DlcsDatabaseFixture dbFixture, ProtagonistAppFactory<Startup> factory)
+    {
+        dlcsContext = dbFixture.DbContext;
+        httpClient = factory.ConfigureBasicAuthedIntegrationTestHttpClient(dbFixture, "API-Test");
+        dbFixture.CleanUp();
+    }
+    
+    [Fact]
+    public async Task Get_CustomHeader_200()
+    {
+        // Arrange
+        const int customerId = 90;
+        var path = $"customers/{customerId}/customHeaders";
+        var customHeader = new CustomHeader()
+        {
+            Id = Guid.NewGuid().ToString(),
+            Customer = customerId,
+            Key = "test-key",
+            Value = "test-value"
+        };
+        
+        await dlcsContext.CustomHeaders.AddAsync(customHeader);
+        await dlcsContext.SaveChangesAsync();
+        
+        // Act
+        var response = await httpClient.AsCustomer(customerId).GetAsync(Path.Combine(path, customHeader.Id));
+        
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+    }
+    
+    [Fact]
+    public async Task Get_CustomHeader_404_IfNotFound()
+    {
+        // Arrange
+        const int customerId = 91;
+        var path = $"customers/{customerId}/customHeaders/{Guid.Empty}";
+        
+        // Act
+        var response = await httpClient.AsCustomer(customerId).GetAsync(path);
+        
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.NotFound);
+    }
+    
+    [Fact]
+    public async Task Delete_CustomHeader_204()
+    {
+        // Arrange
+        const int customerId = 92;
+        var path = $"customers/{customerId}/customHeaders";
+        var customHeader = new CustomHeader()
+        {
+            Id = Guid.NewGuid().ToString(),
+            Customer = customerId,
+            Key = "test-key",
+            Value = "test-value"
+        };
+        
+        await dlcsContext.CustomHeaders.AddAsync(customHeader);
+        await dlcsContext.SaveChangesAsync();
+        
+        // Act
+        var response = await httpClient.AsCustomer(customerId).DeleteAsync(Path.Combine(path, customHeader.Id));
```

Comment:
I think it's cleaner to have the entire path evident in the `path` variable, rather than building it where `httpClient` is called, that way at a quick scan of the tests it's really evident what URL is being tested.

Initially I thought this was testing `DELETE /customers{customerId}/customHeaders` and didn't initially clock the Id was at the end.

e.g.

```cs
[Fact]
public async Task Delete_CustomHeader_204()
{
    // Arrange
    const int customerId = 92;
    var path = $"customers/{customerId}/customHeaders/{customHeader.Id}`

    // other stuff

    var response = await httpClient.AsCustomer(customerId).DeleteAsync(path);
```

```cs
[Fact]
public async Task Delete_CustomHeader_204()
{
    // Arrange
    const int customerId = 92;
    var path = $"customers/{customerId}/customHeaders";

    // other stuff
    var response = await httpClient.AsCustomer(customerId).DeleteAsync(Path.Combine(path, customHeader.Id));
```

Reply:
Agreed. This has now been resolved in 17a80f5a6d1196fde444c6c9a291829124344379